rotkehlchen/db/dbhandler.py

```diff
@@ -2144,46 +2166,47 @@ def delete_data_for_evm_address(
 
     def delete_data_for_evmlike_address(
             self,
-            write_cursor: 'DBCursor',
+            write_cursor: 'DBWriterClient',
             address: ChecksumEvmAddress,
             blockchain: SUPPORTED_EVMLIKE_CHAINS_TYPE,  # pylint: disable=unused-argument
     ) -> None:
         """Deletes all evmlike chain related data from the DB for a single evm address
 
         For now it's always gonna be only zksync lite.
         """
-        other_addresses = self.get_single_blockchain_addresses(
-            cursor=write_cursor,
-            blockchain=SupportedBlockchain.ZKSYNC_LITE,
-        )
-        write_cursor.execute('DELETE FROM used_query_ranges WHERE name = ?', (f'{ZKSYNCLITE_TX_SAVEPREFIX}{address}',))  # noqa: E501
-        other_addresses.remove(address)  # exclude the address in question so it's only the others
-
-        # delete events by tx_hash
-        write_cursor.execute(
-            'SELECT tx_hash, from_address, to_address FROM zksynclite_transactions WHERE '
-            'from_address=? OR to_address=?',
-            (address, address),
-        )
-        hashes_to_remove = [
-            x[0] for x in write_cursor
-            if x[1] not in other_addresses and x[2] not in other_addresses  # pylint: disable=unsupported-membership-test
-        ]
-        for hashes_chunk in get_chunks(hashes_to_remove, n=1000):  # limit num of hashes in a query
-            write_cursor.execute(  # delete transactions themselves
-                f'DELETE FROM zksynclite_transactions WHERE tx_hash IN '
-                f'({",".join("?" * len(hashes_chunk))})',
-                hashes_chunk,
+        with self.conn.read_ctx() as cursor:
+            other_addresses = self.get_single_blockchain_addresses(
+                cursor=cursor,
+                blockchain=SupportedBlockchain.ZKSYNC_LITE,
```

Comment:
just for the indentation... you don't need the read cursor for more than two queries right? Can't you do them first and then operate only with the write cursor?